### PR TITLE
Samantha, section 10

### PR DESCRIPTION
**Pull request by qhxz**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
I didn't count.

Please fill in one error and/or error message you received while working on this assignment.
I don't remember.

What part of the assignment did you spend the most time on?
working out the github side of things.

Comfort Level (1-5):
4 (if 5 is most comfortable)

Completeness Level (1-5):
5 (if 5 is most complete)

What did you think of this deliverable?
ok.

Is there anything in this code that you feel pleased about?
nope.

### CODE REVIEW

**Comment by zqxlts** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
about 2?

Please fill in one error and/or error message you received while working on this assignment.
couldn't get the program to read into the if statements that plays the game.

What part of the assignment did you spend the most time on?
cleaning up the code and possible working out the logic for the korean version

Comfort Level (1-5):
4 (if 5 is most comfortable)

Completeness Level (1-5):
5 (if 5 is most complete)

What did you think of this deliverable?
ok.

Is there anything in this code that you feel pleased about?
nothing in particular

File: script.js [hunk 1]
```diff
@@ -1,4 +1,233 @@
...
+var playCount = 0;
+var winCount = 0;
+var mjpWin = "no one"; // korean SPS winner
+
+// SPS function randomises the computer choice
+function SPS() {
+  var randomiser = Math.floor(Math.random() * 3); // using randomiser to pick a number
+  var computer = ["scissors", "paper", "stone"];
+  return computer[randomiser]; // returning the results of the randomised choice
+}
```
Comment (JustinWong98) [inline script.js:18]: Wow, pretty advanced but good that you used an array to help select the outcomes haha

File: script.js [hunk 2]
```diff
@@ -1,4 +1,233 @@
...
+    playCount += 1;
+    myOutputValue = `<u>${gameMode}</u><br><br>You played ${input}. <br> Computer played ${comPlay}. <br><br>You lose! Try again?<br><br>${username} has won ${winCount}/${playCount} times against the computer.`;
+  }
+  return myOutputValue;
+}
+
+// ======================= MAIN FUNCTION ========================
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  input = input.toLowerCase();
```
Comment (JustinWong98) [inline script.js:195]: Wow, smart move to standardize the input by making sure its all lower case!

File: script.js [hunk 3]
```diff
@@ -1,4 +1,233 @@
...
+  if (gameMode == "Muk-Jji-Pa!") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = koreanSPS(comPlay, input);
+  }
+  if (gameMode == "Computer vs Computer") {
+    console.log("game mode: " + gameMode);
+    myOutputValue = comVcom(comPlay, input);
+  }
+  return myOutputValue;
+}
```
Comment (JustinWong98) [inline script.js:39]: Good job with using helper functions for each different mode!

File: script.js [hunk 4]
```diff
@@ -1,4 +1,233 @@
...
+    (input == "scissors" && comPlay == "stone") ||
+    (input == "paper" && comPlay == "scissors") ||
+    (input == "stone" && comPlay == "paper")
+  ) {
+    // +1 play count each time the game runs
+    playCount += 1;
+    myOutputValue = `You played ${input}. <br> Computer played ${comPlay}. <br><br>You lose! Try again?<br><br>${username} has won ${winCount}/${playCount} times against the computer.`;
+  }
+  return myOutputValue;
+}
```
Comment (JustinWong98) [inline script.js:76]: Since playCount increases by 1 in every if statement, consider putting it before the if statements to reduce redundancy.

**Review by JustinWong98 — COMMENTED** [review]
Very good! Great use of pseudocode and refactoring and I was pleasantly surprised to see an array being used to help select the computer choice :) Keep it up!

**Pull request closed without merging**